File: Examples/CSharp/AsposePDF/Working-Document/OptimizeDocument.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2013 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;

namespace CSharp.AsposePdf.WorkingDocuments
{
    public class OptimizeDocument
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdf_WorkingDocuments();

            //open document
            Document pdfDocument = new Document(dataDir + "OptimizeDocument.pdf");

            //optimize for web
            pdfDocument.Optimize();

            //save output document
            pdfDocument.Save(dataDir + "OptimizeDocument_out.pdf");
 
        }
    }
}

File: Examples/VisualBasic/AsposePDF/Working-Document/OptimizeDocument.vb
Imports System.IO
Imports System
Imports Microsoft.VisualBasic
Imports Aspose.Pdf

Namespace AsposePDF.WorkingDocuments
    Public Class OptimizeDocument
        Public Shared Sub Run()
            ' ExStart:OptimizeDocument
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdf_WorkingDocuments()

            ' Open document
            Dim pdfDocument As New Document(dataDir & Convert.ToString("OptimizeDocument.pdf"))

            ' Optimize for web
            pdfDocument.Optimize()

            dataDir = dataDir & Convert.ToString("OptimizeDocument_out_.pdf")

            ' Save output document
            pdfDocument.Save(dataDir)
            ' ExEnd:OptimizeDocument
            Console.WriteLine(Convert.ToString(vbLf & "Document optimized successfully for web." & vbLf & "File saved at ") & dataDir)
        End Sub
    End Class
End Namespace

